packages/cli/src/cmds/validator/keymanager/decryptKeystores/worker.ts

```diff
@@ -1,9 +1,29 @@
 import fs from "node:fs";
+import worker from "node:worker_threads";
 import {expose} from "@chainsafe/threads/worker";
 import {Transfer, TransferDescriptor} from "@chainsafe/threads";
 import {Keystore} from "@chainsafe/bls-keystore";
 import {DecryptKeystoreArgs, DecryptKeystoreWorkerAPI, isLocalKeystoreDefinition} from "./types.js";
 
+/**
+ * @param buffer The ArrayBuffer to be returned as transferable
+ * @returns a buffer that can be transferred. If the provided buffer is marked as untransferable, a copy is returned
+ */
+function transferableArrayBuffer(buffer: ArrayBuffer): ArrayBuffer {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
+  const unknownWorker = worker as any;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+  const isMarkedAsUntransferable = unknownWorker["isMarkedAsUntransferable"];
+  // Can be updated to direct access once minimal version of node is 21
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+  if (isMarkedAsUntransferable && isMarkedAsUntransferable(buffer)) {
```

Comment:
Isn't the buffer always untransferable or why do we need this check? Based on https://github.com/nodejs/node/pull/47604 it would previously (node 20) just silently copy the buffer while with node 21 it throws an error. The problem seems to be related to @chainsafe/bls-keystore and how the keystore buffer is created.

Upgrading @chainsafe/bls-keystore fixes the issue for me since with v3 it uses Unit8Array instead of Buffer. Will do some performance testing with new version and open PR for that

Reply:
Right, it might not be transferred at all with node20.

If the new `@chainsafe/bls-keystore` offers equivalent or better perf that's definitely the better option.

Reply:
Could you try https://github.com/ChainSafe/lodestar/pull/6253 on your machine? At least on linux there are no issues running unit tests on node 21

Reply:
I can confirm that it works on osx with node 21